### ISSUE TITLE
Database Content

### ISSUE THREAD

**Issue by Gollwu** [Gollwu/LoLManager#44]
Hey,

It could be nice to have a first full database schema so that we can create a first snapshot of the database (even though we can amend it later). I'll take care of the creation of the tables but I need everyone's help to know what kind of attributes we need on each entity. Mostly we need the guy from the formulas here :)

**Comment by Gollwu** [comment]
Let's discuss table by table : 

Table champion

name   

A Value for each "role" (To be discussed, it seems weird to me as a support can be tank for example... Maybe Support tank and support AP?)
ADCarry
Tank
ADTank
APTank
APCarry
AssassinAP
AssassinAD
Support

Values to determine the champion strenghts 
Ganking
Jungling (understand sustain in the jungle and camp clearing)
Roaming
WaveClear
Siege
Engage
Disengage
CrowdControl

Values to determine if a champion fits into a strategy
Early
MidGame
LateGame
Splitpush
TeamFight
More?

We also need to discuss if it is okay to have a database full of stats even tho we don't use it :)

**Comment by Nhacsam** [comment]
I propose to create a table Role and a table ChampionRoleAffinity with : 
role: Role
champion: Champion
affinity: integer % bettween 0 and 100

**Comment by Gollwu** [comment]
Yeah that's a possibility. You mean affinity for roles (ADCarry ...) and Strategies (Splitpushing, early.. etc)?. I don't see what it brings to split the table in 3 ?

**Comment by Nhacsam** [comment]
We can use the same role table for player affinity. In matchmaking, when we choose the player's role, it will be easier to choose the champ he will use.

**Comment by Gollwu** [comment]
Well if you have a clear idea why it should be separated for data-sending to the client I have no objection. Anyway, anything to say about the amount of attributes and their nature? Missing something?

**Comment by Sowerdski** [comment]
These are pretty good ideas indeed. I would see something like one table for each player which would contain a subtable with stats that would never depend on the champion, the role, or the strategy in use (like mechanics etc..), this Player table would also contain a sub-table to define its abilities with every champion, another for every strategy, and another for every role (i don't know if i explained it well)                

Then we would have a Champion table that would contain every stat that has no link with player stats, or in a simpler way, the champion characterics (as Gollwu said above : ganking, jungling roaming etc... and even strategy fitting)

In a Nutshell :
Database:
       - Champion Table
       - Player Table :
                 - Player SubTable
                 - Champion affinity SubTable
                 - Role Affinity SubTable
                 - Strategy Affinity SubTable

**Comment by Sowerdski** [comment]
Currently, player-reliant statistics are (according to the BDD Params doc file) : 
- Mechanics
- Laning
- Farming
- Ganking
- Smiting
- Warding
- In Game Attitude
- Charisma / Personnality (Because it's SWAG)
- Role Affinity (Designed as a subtable? => cf. post above / This stats includes Champion Affinity!!!)

To synthesize all previous posts, these stats should also be added:
- Strategy Affinity

Moreover, Champion affinity shall be distinguished from Role affinity as called in the original Database description document .

**Comment by Sowerdski** [comment]
Now, let's focus a bit on the champion table. I agree with Gollwu about the value describing the "role" of a champion, however, I would organize this in a different way :
- Damage Source (e.g: 0 for physical, 1 for magical, 2 for hybrid)
- Damage Amount (e.g: 0 for support or tank, 1 for carries (dps), 2 for assasins (burst))
- Tanking Amount (e.g: 0 for carries/assassins/ap supports, 1 for Ad tanks, ap tanks, 2 for full tanks)

Next issue : champions strengths. It would be better to choose values in order to maintain a certain level of relevancy regarding the player table. We could think of something like this : 
- Ganking ( link w/ Player Table (PT) ganking stat )
- Jungling ( link w/ PT farming stat but processed in game for the jungler only)
- Strategy Affinity Table (actually, fitting into a strategy give information about whether or not the champ is a good pusher etc... , this allows us to lessen the number of stats used)

Feel free to discuss that proposal as I may have missed some essential characteristics.

N.B: Please note that the values are only examples given, I will think about a efficient way to bring these stats into formulas.

**Comment by Gollwu** [comment]
On vire Krako?

**Comment by fabienDaou** [comment]
I feel like we (or at least me) do not have a clear idea of what we want to simulate (which behaviours...). We are making the same mistakes all over again (start coding then we feel like it is clunky finally we change everything bacause it is crap). Do we want to simulate matchups? Do we want to simulate split, teamfight strats? Do we want to take into account champions statistics? Do we want to translate the ability of a player to play a certain champ?...

Once we answered all these questions, we can work on a schema that fits our needs. 

We are just lazy retards (especially ElJefe) (dont hijack my comment plz jefe)

**Comment by Gollwu** [comment]
I AM THE KING OF COMMENTS

**Comment by qr7hur** [comment]
Fabien is right, we are just tardy lizards.

**Comment by Gollwu** [comment]
Krako t'es viré 

![vire](https://cloud.githubusercontent.com/assets/6659146/6928727/6a680096-d7f6-11e4-9e5c-6d63df2bf0b3.png)

**Comment by Gollwu** [comment]
Hey : big discussion here for everything, feel free to participate (french only)

https://docs.google.com/document/d/11wo59iOtsIWaIkql1g84uQSXRxDGBZ5QzVM6tVhVoHc/edit

Remember it's just a brainstorming :)

**Comment by fabienDaou** [comment]
I added a comment in the google doc. I do not know if you are notified.

**Comment by Gollwu** [comment]
Yup it seems like a nice correction to me. Still need to to take into account the match up though.

What do you think about the other features? (super porwers of champions, reddit page, the things to be calculated for the management system...)

**Comment by fabienDaou** [comment]
Super Powers of champions are really cool. Nonetheless, I would not focus on this while we are trying to settle the basics of laning, teamfight.. I would suggest to develop something generic first. Then it will be easier to integrate these superpowers. If we develop clean, we can easily add modules corresponding to these behaviours.

Same for the reddit page I guess. This would be the next iteration. 

(I added several comments in the google doc)

**Comment by Sowerdski** [comment]
I am currently wrting a first version of the entire fight simulation, based essentially on the brainstorming we did earlier, this may take some time though. Once it is completed, we can discuss every point to update the document progressively (I thought It was better to get the whole thing in a first place, even though it wouldn't be perfectly accurate)

You can read the document here (ofc not completed yet):
https://docs.google.com/document/d/1E0kQAi-EFUmWQj83m01s2Kpaiza6UaXIcDI9DG8iVe8/edit?usp=sharing

**Comment by Gollwu** [comment]
The approach on the engage is fine. However, I disagree with one point: The players can fail and still engage even if they shouldn't (If they are bronze or if they have poor shotcalling or smthg like that). 

Rest seems okay, een tho I didn't recheck every formula directly, but it's not the point
